script.rb

```diff
@@ -0,0 +1,152 @@
+module Enumerable
+  # -----------my_each methods--------------
+
+  def my_each
+    return to_enum unless block_given?
+
+    index = 0
+    arr = to_a
+    while index < arr.length
+      yield(arr[index])
+      index += 1
+    end
+    self
+  end
+
+  # -----------my_each_with_index----------------------
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = to_a
+    index = 0
+    while index < arr.length
+      yield(arr[index], index)
+      index += 1
+    end
+    self
+  end
+
+  # -----------my_select----------------------
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    arr = to_a
+    arr.my_each { |x| selected.push(x) if yield(x) }
+    selected
+  end
+
+  #--------------------my_all-----------------------
+  def my_all?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return false unless !i[args].nil? || i[args] == 1 } # if Regexp
+      else
+        my_each { |i| return false unless i == args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return false unless i.is_a?(args) }
+
+    # if not Regexp
+
+    elsif args.nil? && block_given?
+      my_each { |i| return false unless yield i }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
+  #-------------------my_any?-----------------------
+
+  def my_any?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return true if !i[args].nil? || i[args] == 1 } # if Regexp
+      else
+        my_each { |i| return true if i == args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return true if i.is_a?(args) }
+
+    # if not Regexp
+
+    elsif args.nil? && block_given?
+      my_each { |i| return true if yield i }
+    else
+      my_each { |i| return false unless i.nil? || i == false }
+    end
+    false
+  end
+
+  #-------------------my_none?-----------------------
+  def my_none?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return false unless i[args].nil? || i[args] != 1 } # if Regexp
+      else
+        my_each { |i| return false unless i != args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return false if i.is_a?(args) }
+
+    elsif args.nil? && block_given?
+      my_each { |i| return false if yield i }
+    else
+      my_each { |i| return false unless i.nil? || i != false }
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false].my_none? == [nil, false].none? => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/93888608-76bb2100-fca5-11ea-8766-291d43929de6.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/93888644-82a6e300-fca5-11ea-8554-898898a3b16c.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/d/) == %w[dog door rod blade].none?(/d/) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/93888944-dca7a880-fca5-11ea-814e-f193f866dd78.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/93888993-eb8e5b00-fca5-11ea-8e31-973aef90633c.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?